### PR TITLE
Dsg 2645/address sonarqube code smells

### PR DESCRIPTION
**Pull request by lucywang-nv**
# Basics

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Styling and layout changes
- [x] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?
Ticket: https://dol-ewf.atlassian.net/browse/DSG-2645 
Address the code smells found on Sonar Qube.

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

**Review by thechechogomez — APPROVED** [review]
LGTOM

File: libs/agency/feature/dashboard/src/lib/components/events-log/event-types/transaction-data-changed/transaction-list-data-changed/transaction-list-data-changed.component.ts [hunk 1]
```diff
@@ -35,6 +35,57 @@ export class TransactionListDataChangedComponent implements OnInit {
...
+      if (!valid) continue;
+
+      let _newStateValue = newStateValue;
+      let newDocumentId = '';
+      const _oldStateValue = '';
+      const oldDocumentId = '';
+
+      if (component && component.type === 'nuverialMultipleFileUpload') {
+        newDocumentId = newStateValue;
+        _newStateValue = this._fileDisplayValue(index); // Doesn't yet support file lists within lists
```
[the remark this answers is not in the document]
Comment (kevincheng0) [reply libs/agency/feature/dashboard/src/lib/components/events-log/event-types/transaction-data-changed/transaction-list-data-changed/transaction-list-data-changed.component.ts:52]: I don't think we are ready to handle form lists yet, there is no story as far as I know

File: libs/shared/ui/nuverial/src/lib/directives/date-mask/date-mask.directive.ts [hunk 2]
```diff
@@ -39,44 +39,46 @@ export class NuverialDateMaskDirective {
...
+    switch (slashesCount) {
+      case 0:
+        if (value.length > 2) {
+          const nextValue = `${value.slice(0, 2)}/${value.slice(2)}`;
+
+          return this._formatToDate(nextValue);
+        }
+        break;
+      case 1:
+        if (value.startsWith('/')) {
```
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/directives/date-mask/date-mask.directive.ts:57]: Does this change the result? did you make sure to test in the browser?

File: libs/shared/ui/nuverial/src/lib/components/tree/tree.component.scss [hunk 3]
```diff
@@ -173,10 +173,6 @@ $end-line-height: calc($node-height + $gap-size);
       position: relative;
     }
 
-    ul.sublist li {
```
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/components/tree/tree.component.scss:176]: Was this duplicate?

File: libs/shared/ui/nuverial/src/lib/components/footer-actions/footer-actions.component.ts [hunk 4]
```diff
@@ -78,7 +78,7 @@ export class NuverialFooterActionsComponent implements OnDestroy {
     this._renderer.addClass(document.body, FOOTER_ACTIONS_OPEN_CLASS);
 
     if (actions.length <= 3) {
-      this.buttonActionsList = this._applyButtonStylingToActions(actions.reverse());
+      this.buttonActionsList = this._applyButtonStylingToActions([...actions].reverse());
```
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/components/footer-actions/footer-actions.component.ts:81]: Doesn't this change the result?

**Review by thechechogomez — APPROVED** [review]

File: libs/shared/data-access/work-api/src/lib/models/form-configuration/form-configuration.model.ts [hunk 5]
```diff
@@ -60,23 +60,11 @@ export const removeEmptyProperties = (obj: ExtendedComponentSchema): any => {
...
-        if (FORM_CONFIGURATION_KEYS_TO_REMOVE.includes(key)) {
-          return acc;
-        }
-
-        if (acceptValueType(value)) {
+        if (
+          FORM_CONFIGURATION_KEYS_TO_KEEP.includes(key) ||
+          (FORM_CONFIGURATION_KEYS_WITH_EMPTY_VALUES_TO_KEEP.includes(key) && !acceptValueType(value)) ||
+          (!FORM_CONFIGURATION_KEYS_TO_REMOVE.includes(key) && acceptValueType(value))
+        ) {
```
Comment (davidyoung77) [inline libs/shared/data-access/work-api/src/lib/models/form-configuration/form-configuration.model.ts:67]: Does it complain if we do it like this?, I think this is a little easier to read.

```suggestion
       if (FORM_CONFIGURATION_KEYS_TO_REMOVE.includes(key)) {
          return acc;
        }
        
        if (
          FORM_CONFIGURATION_KEYS_TO_KEEP.includes(key) ||
          (FORM_CONFIGURATION_KEYS_WITH_EMPTY_VALUES_TO_KEEP.includes(key) && !acceptValueType(value)) ||
          (acceptValueType(value))
        ) {
```

**Review by davidyoung77 — COMMENTED** [review]
Nice work I have a few comments

**Pull request merged**